scala/runtime/src/main/scala/org/coursera/courier/templates/ScalaEnumTemplate.scala

```diff
@@ -41,11 +41,24 @@ abstract class ScalaEnumTemplate[T <: ScalaEnumTemplateSymbol]
   /**
    * Schema properties defined on this enum, if any.
    */
-  lazy val properties: Option[DataMap] = {
-    Option(SCHEMA.getProperties.get(symbolProperties)).collect {
-      case enumProps: DataMap => enumProps
+  def properties: Option[DataMap] =
+  // Implementation note: using a lazy field can result in deadlock.
```

Comment:
[edit]
I see you've explained deadlock can occur when initializing objects with the class loader. A comment that references the test, where you've documented in detail, would be helpful. 

[original comment below]

I don't doubt that this is true given the effort you've put in to land this, but it's not obvious to me why this particular lazy val is problematic. I read [this article](https://blog.codecentric.de/en/2016/02/lazy-vals-scala-look-hood/) to gather some context, and the deadlock scenario they describe is due to an initialization chain that crosses synchronization boundaries. Is there a similar cause here? 

Reading the code, initialization seems pretty innocent, doesn't smell like a 'dangerous' lazy usage. Hoping to understand better, as if I'm ignorant here, we've probably got lots of code floating around where laziness introduces the possibility of a similar failure mode.

Reply:
There are two lazy vals and a class lock, the other lazy val is in courscala project I think ... The article you linked to is good: _we can easily get into the same situation by using other concurrency and synchronization primitives_ Our other synchronization primitive is in class loading, somewhere.